lib/JSON/Validator/OpenAPI/Mojolicious.pm

```diff
@@ -225,6 +246,83 @@ sub _coerce_by_collection_format {
   return $single ? $data->[0] : $data;
 }
 
+sub _coerce_object_by_style {
+  my ($self, $c, $data, $p) = @_;
+
+  my $def     = $DEFAULT_SERIALIZATION{$p->{in}} || {};
+  my $style   = $p->{style}   // $def->{style};
+  my $explode = $p->{explode} // $def->{explode};
+  return $data unless $style;
+
+  # Special serializations
+  if ($style eq 'form' && $explode) {
+    return $c->req->url->query->to_hash;
+  }
+  elsif ($style eq 'deepObject') {
+    return $self->_coerce_deep_object($c, $p);
+  }
+
+  return unless defined $data;
+
+  if ($explode) {
+    my $re = $OBJ_STYLE_RE{$style}->{true} or return $data;
+    return if $style eq 'matrix' && $data !~ s/^;//;
+    return if $style eq 'label'  && $data !~ s/^\.//;
+    my $params = Mojo::Parameters->new;
+    $params->append(Mojo::Parameters->new($_)) for split /$re/, $data;
+    return $params->to_hash;
+  }
+  else {
+    my $re = $OBJ_STYLE_RE{$style}->{false} or return $data;
+    return if $style eq 'matrix' && $data !~ s/^;\Q$p->{name}\E=//;
+    return if $style eq 'label'  && $data !~ s/^\.//;
+    return Mojo::Parameters->new->pairs([split /$re/, $data])->to_hash;
+  }
+}
+
+sub _coerce_deep_object {
+  my ($self, $c, $p) = @_;
+
+  my $result;
+
+  my @pairs = @{$c->req->params->pairs};
+  while (my ($k, $v) = splice(@pairs, 0, 2)) {
```

Comment:
I think this while-loop is very hard to read, with all the `$last_ref` stuff and `($#{$$last_ref} + 1)`. I'm going to take the PR anyways, since I'm trying to clean up my backlog after being preoccupied for too long.

If you find time to clean it up, then I will for sure take that PR ❤️